pkg/autodiscovery/autoconfig.go

```diff
@@ -61,28 +63,32 @@ type providerDescriptor struct {
 // Notice the `AutoConfig` public API speaks in terms of `integration.Config`,
 // meaning that you cannot use it to schedule integrations instances directly.
 type AutoConfig struct {
-	providers         []*providerDescriptor
-	templateCache     *TemplateCache
-	listeners         []listeners.ServiceListener
-	configResolver    *ConfigResolver
-	configsPollTicker *time.Ticker
-	scheduler         scheduler.Scheduler
-	stop              chan bool
-	pollerActive      bool
-	health            *health.Handle
-	store             *store
-	m                 sync.RWMutex
+	providers          []*providerDescriptor
+	templateCache      *TemplateCache
+	listeners          []listeners.ServiceListener
+	listenerCandidates map[string]listeners.ServiceListenerFactory
+	listenerRetryStop  chan struct{}
+	configResolver     *ConfigResolver
+	configsPollTicker  *time.Ticker
+	scheduler          scheduler.Scheduler
+	pollerStop         chan struct{}
+	pollerActive       bool
+	health             *health.Handle
+	store              *store
+	m                  sync.RWMutex
 }
 
 // NewAutoConfig creates an AutoConfig instance.
 func NewAutoConfig(scheduler scheduler.Scheduler) *AutoConfig {
 	ac := &AutoConfig{
-		providers:     make([]*providerDescriptor, 0, 5),
-		templateCache: NewTemplateCache(),
-		stop:          make(chan bool),
-		store:         newStore(),
-		health:        health.Register("ad-autoconfig"),
-		scheduler:     scheduler,
+		providers:          make([]*providerDescriptor, 0, 5),
+		listenerCandidates: make(map[string]listeners.ServiceListenerFactory),
+		listenerRetryStop:  nil, // We'll open it if needed
```

Comment:
I guess we can remove this line if we set it to `nil` right?

Reply:
That's a comment for the future 33-year-old me that'll wonder why we don't initialise it here ;)